### PR TITLE
Calendar sync

### PR DESCRIPTION
**Pull request by Nouuu**
La meilleure fonctionnalité du CLI :-)

C'est la v1, des changements viendront peut être enrichir la fonctionnalité !

### CODE REVIEW

File: src/campus-location.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  ['MONTSOURIS', ['5 rue Lemaignan, 75014 Paris', '3']],
+  ['MONTROUGE', ['11 rue Camille Pelletan, 92120 Montrouge', '6']],
+  ['JOURDAN', ['6-10 bd Jourdan 75014 Paris', '7']],
+  ['VAUGIRARD', ['273-277 rue de Vaugirard, 75012 Paris', '9']],
+  ["MAIN D'OR", ['8‐14 Passage de la Main d’Or 75011 Paris', '8']],
+]);
+
+export const getCampusLocation = (key: string): string[] => {
+  return campusLocations.get(key) || [undefined, '11'];
+};
```
[the remark this answers is not in the document]
Comment (Nouuu) [reply src/campus-location.ts:13]: resolved

File: README.md [hunk 2]
```diff
@@ -62,3 +66,40 @@ You can also join or quit a project group.
...
+
+* Go to [https://developers.google.com/calendar/quickstart/php](https://developers.google.com/calendar/quickstart/php)
+* Create a Google Calendar API project
+
+|                                           |                                           |
+| ----------------------------------------- | ----------------------------------------- |
+| ![image](https://i.imgur.com/xZkQC03.png) | ![image](https://i.imgur.com/QVQ6vH2.png) |
+| ![image](https://i.imgur.com/AmHIOfb.png) |                                           |
+
+* Save the `credentials.json` file at the root of the project:
```
[the remark this answers is not in the document]
Comment (Nouuu) [reply README.md:88]: work in progress but in good way => resolved but need some tests before validate

[the remark this answers is not in the document]
Comment (Nouuu) [reply README.md:88]: work in progress but in good way => resolved but need some tests before validate

File: README.md [hunk 3]
```diff
@@ -62,3 +66,40 @@ You can also join or quit a project group.
...
+
+To retrieve your calendar id:
+
+* Go to https://calendar.google.com
+* Go to "Settings and sharing" of the calendar dedicated to your planning
+* Go to "Settings and sharing" of the calendar dedicated to your planning
+![image](https://i.imgur.com/QAZPssf.png)
+* You will find the id of your diary in the section "Integrate the calendar"
+![image](https://i.imgur.com/1p0Ra2q.png)
+* Save this id in `.env` file at the root fo the project
```
[the remark this answers is not in the document]
Comment (Nouuu) [reply README.md:104]: work in progress but in good way => resolved but need some tests before validate

File: src/google-calendar.ts [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+            }
+          }, 100);
+        },
+      );
+    })
+    .finally(() => {
+      progressBar.stop();
+      eventAdded = true;
+      taskComplete();
+    });
```
[the remark this answers is not in the document]
Comment (Nouuu) [reply src/google-calendar.ts]: resolved but need some tests to ensure `setTimeout()` is respected

File: src/google-calendar.ts [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+      timeZone: 'Europe/Paris',
+    },
+    end: {
+      dateTime: formatRFC3339(new Date(agendaItem.end_date)),
+      timeZone: 'Europe/Paris',
+    },
+  };
+};
+
+const getClient = (credentials): OAuth2Client => {
```
[the remark this answers is not in the document]
Comment (Nouuu) [reply src/google-calendar.ts]: now it has !

**Comment by quantumsheep** [comment]
@Nouuu can you fix the conflicts?

**Pull request merged**